crates/evm/src/metrics.rs

```diff
@@ -2,14 +2,41 @@
 //!
 //! Block processing related to syncing should take care to update the metrics by using either
 //! [`ExecutorMetrics::execute_metered`] or [`ExecutorMetrics::metered_one`].
-use std::time::Instant;
-
+use crate::{execute::Executor, system_calls::OnStateHook};
 use metrics::{Counter, Gauge, Histogram};
 use reth_execution_types::{BlockExecutionInput, BlockExecutionOutput};
 use reth_metrics::Metrics;
 use reth_primitives::BlockWithSenders;
+use revm_primitives::ResultAndState;
+use std::time::Instant;
+
+/// Wrapper struct that combines metrics and state hook
+struct StateHookWrapper {
```

Comment:
can we rename this to smth like `MeteredStateHook`?

Reply:
good call, done here https://github.com/paradigmxyz/reth/pull/12316/commits/324d4ff5d5568b83ac285c17a4ef49f463cdcd57